packages/openapi-generator/src/file-serializer/schema.ts

```diff
@@ -0,0 +1,103 @@
+import { codeBlock } from '@sap-cloud-sdk/util';
+import {
+  OpenApiSchema,
+  OpenApiObjectSchema,
+  OpenApiObjectSchemaProperty
+} from '../openapi-types';
+import { getType } from '../parser/type-mapping';
+import {
+  isReferenceObject,
+  parseTypeNameFromRef,
+  isArraySchema,
+  isObjectSchema,
+  isEnumSchema,
+  isOneOfSchema,
+  isAllOfSchema,
+  isAnyOfSchema,
+  isNotSchema
+} from '../model';
+
+/**
+ * Serialize a schema.
+ * @param schema Parsed schema to be serialized.
+ * @returns The serialized schema
+ */
+export function serializeSchema(schema: OpenApiSchema): string {
+  if (isReferenceObject(schema)) {
+    return parseTypeNameFromRef(schema);
+  }
+  if (isArraySchema(schema)) {
```

Comment:
Architecture question:
Does it make sense to define a serializer interface which standardises all shema types and related serializing methods. In this method one would than just loop over all check and method pairs (isSomething,serialiazeSomething). Doing so we could potentially serialize letter to different languages?

Reply:
I created a ticket for this.